src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleaner.py

```diff
@@ -327,6 +328,8 @@ def _dispatchWflow(self, wflow):
             self.logger.info(msg)
             self._checkStatusAdvanceExpired(wflow, additionalInfo=msg)
         elif wflow['RequestStatus'] in ['announced', 'rejected', 'aborted-completed']:
+            if wflow['RequestStatus'] == 'announced' and not wflow['ParentageResolved']:
+                msg = f"Adding workflow: {wflow['RequestName']} - 'ParentageResolved' flag set to false."
```

Comment:
I do not see the point of those two lines here.

Reply:
Yes, maybe moving it down to the archival method would be good enough. I think it's important to log that A workflow can't go through the component (completely) because its parentage has not been resolved.